docs/operator-guide/upgrade-edp-3.9.md

```diff
@@ -0,0 +1,90 @@
+      2. [Integrate Capsule](https://epam.github.io/edp-install/operator-guide/capsule) with EDP platform.
+      3. Update edp-install values file:
+
+      ``` yaml title="values.yaml"
+      ...
+      cd-pipeline-operator:
+        tenancyEngine: "capsule"
+      ...
+      ```
+
+  === "Disable tenancy engine"
+
+      1. To disable tenancy engine update edp-install values file:
+
+      ``` yaml title="values.yaml"
+      ...
+      cd-pipeline-operator:
+        tenancyEngine: "none"
+      ...
+      ```
+
+2. (Optional) Align Keycloak integration.
+
+  In KubeRocketCI version 3.9.0, the Keycloak configuration procedure has been altered in the [values](https://github.com/epam/edp-install/blob/v3.9.0/deploy-templates/values.yaml#L461) file. Please be attentive and align the configuration for your own use case.
+
+  === "Values 3.8.1"
+
+      ``` yaml title="values.yaml"
+      sso:
+        # -- Install OAuth2-proxy and Keycloak CRs as a part of EDP deployment.
+        enabled: false
+        # -- Defines Keycloak realm name that is used as the Identity Provider (IdP) realm
+        realmName: "broker"
+        # -- Keycloak URL.
+        keycloakUrl: https://keycloak.example.com
+      ```
+
+  === "Values 3.9.0"
+
+      ``` yaml title="values.yaml"
+      sso:
+        # -- Install OAuth2-proxy and Keycloak CRs as a part of EDP deployment.
+        enabled: false
+
+        keycloakOperatorResources:
+          # Set to false if using the add-ons approach (refer to https://github.com/epam/edp-cluster-add-ons)
+          # for EDP installation and if the extension-oidc is already installed.
+          # This prevents the creation of an additional Keycloak resource and secret.
+          # The 'kind' and 'name' must be specified in case of using an existing Keycloak/ClusterKeycloak resource.
+          # Create kind: Keycloak as a part of chart installation
+          createKeycloakCR: true
+          # Can be Keycloak or ClusterKeycloak.
+          kind: Keycloak
+          # Name of kind: Keycloak/ClusterKeycloak CR.
+          name: main
+
+        # -- Defines the Keycloak realm name, which by default is named after the namespace where EDP is deployed.
+        # realmName: edp
+        # -- Defines Keycloak sso realm name that is used as the Identity Provider (IdP) realm
+        ssoRealmName: "broker"
+        # -- Defines Keycloak client name that is used for the Identity Provider (IdP) client
+        ssoClientName: "edp"
+        # -- Keycloak URL.
+        keycloakUrl: https://keycloak.example.com/auth
+      ```
+
+3. To upgrade EDP to the v3.9.0, run the following command:
+
+  ```bash
+  helm upgrade edp epamedp/edp-install -n edp --values values.yaml --version=3.9.0
+  ```
+
+  !!! Note
+      To verify the installation, it is possible to test the deployment before applying it to the cluster with the `--dry-run` tag:<br>
```

Comment:
To verify the installation, test the deployment before applying it to the cluster with the `--dry-run` tag:<br>